File: PortProject.Api/Application/Dock/DTOs/DockDto.cs
namespace PortProject.Api.Application.Dock.DTOs
{
    public class DockDto
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string LocationZone { get; set; }
        public string LocationSection { get; set; }
        public double LengthInMeters { get; set; }
        public double DepthInMeters { get; set; }
        public double MaxDraftInMeters { get; set; }
        public int NumberOfSTSCranes { get; set; }
        public List<string> AllowedVesselTypeIds { get; set; } = new();
    }
}

File: port-spa-app/src/infrastructure/repositories/dock/dock.dto.ts
// DTOs (Data Transfer Objects) for Dock
// These define the exact shape of data sent to and received from the API

export interface DockCreateDto {
    id: string;
    name: string;
    locationZone: string;
    locationSection: string;
    lengthInMeters: number;
    depthInMeters: number;
    maxDraftInMeters: number;
    numberOfSTSCranes: number;
    allowedVesselTypeIds?: string[];
}

export interface UpdateDockDto {
    name: string;
    locationZone: string;
    locationSection: string;
    lengthInMeters: number;
    depthInMeters: number;
    maxDraftInMeters: number;
    numberOfSTSCranes: number;
    allowedVesselTypeIds?: string[];
}

